### ISSUE TITLE
JetBrains: LLM name shows "Default" when opening older Chat and restarting the editor fixes the issue and showing "Claude 2.0 by Anthropic".

### ISSUE THREAD

**Issue by jay-fibi** [sourcegraph/jetbrains#997]
### Installation Information

Plugin version: `5.3.3914-nightly`
IDE version: `IC-233.14475.28`

### Describe the bug

**Steps to reproduce:**

1. Install Cody plugin `v5.3.1544-nightly` in your JetBrains IntelliJ IDEA editor
2. Delete the `"RunOnceActivity.CodyHistoryLlmMigration": "true"` from `workspace.xml`
3. Create multiple chats with different LLM
4. having THESE changes run the same project
5. Upgrade to` v5.3.3914-nightly` and open Cody
6. Go to "Chat History" and open older chats i.e. last week 

### Observed behavior:
Opening a chat shows LLM name as "Default" in LLM dropdown.

### Addition checks:
Close the editor and relaunch it and check these chats from step-6. It will now show "Claude 2.0 by Anthropic".

### Expected behavior

It should show "Claude 2.0 by Anthropic" without requiring restart.

### Additional context

**Video link:**

https://drive.google.com/file/d/1n9TqgRB4QuzCijhWGa3lR6Zz1edwUZoJ/view?usp=drive_link

**Comment by mkondratek** [comment]
Hi @jay-fibi 👋 in this case the testing seems to be a bit disorganized - what I think happened here is that you see "Default" for some chat that has been created and and badly migrated already (so we cannot resolve the model name for it anymore). 

Note that for all the chats created in the testing session the name is correct - that is important and enough.

---

I am going to try to explain you what happens b/w versions:

`v5.3.1544-nightly` - in this version we store the LLM as it's display name e.g. `"Claude 2.0 by Anthropic"`.  

We noticed that this approach is poor as we need the model "key" (the value that we can use in the backend). Hence, we decided to introduce a new persistence for LLMs.

The new approach persists three values: `model` (which is the key - useful for the backend), `provider` (e.g. "Anthropic" or "OpenAI", etc), and `title` (e.g. "Claude 2.0" or "GPT-4 Turbo Preview", etc).

Since we change the persistence the way we persists LLMs we must **migrate** the old values. 

The new persistence and the migration has been introduced in `5.3.2999-nightly`. The migration had two issues
1️⃣ It was **removing the old field** in persistence (the display name field). 
2️⃣ It was **incorrectly handling** two models (one for enterprise and one for pro)
These two created a scenario when we were not able to migrate the llm persistence but removed the old field anyway. 

Both these issues are fixed now by https://github.com/sourcegraph/jetbrains/pull/990 and released in `5.3.3914-nightly`.

---

tl'dr

1. The chat that gives you "Default" now was created in `v5.3.1544-nightly`.
2. Then you run `5.3.2999-nightly` - the migration was performed and due to the bugged behavior it removed LLM from persistence.
3. `5.3.3914-nightly` cannot fix that chat anymore since the former field in persistence has been removed by ☝️
